File: batches_csharp/batch_10_code_reordering/ref.cs
using System;

class Stats {
    static (double avg, int max, int min) Compute(int[] data) {
        int mx = data[0], mn = data[0], total = 0;
        foreach (int v in data) {
            if (v > mx) mx = v;
            if (v < mn) mn = v;
            total += v;
        }
        double avg = data.Length > 0 ? (double)total / data.Length : 0;
        return (avg, mx, mn);
    }

    static void Main() {
        int[] d = {10, 20, 30, 40, 50};
        var r = Compute(d);
        Console.WriteLine($"avg={r.avg:F1} max={r.max} min={r.min}");
    }
}

File: batches_cpp/batch_10_code_reordering/ref.cpp
#include <iostream>
#include <vector>
using namespace std;

void computeStats(const vector<int>& data, double& avg, int& mx, int& mn) {
    mx = data[0]; mn = data[0];
    int total = 0;
    for (int v : data) {
        if (v > mx) mx = v;
        if (v < mn) mn = v;
        total += v;
    }
    avg = !data.empty() ? (double)total / data.size() : 0;
}

int main() {
    vector<int> d = {10, 20, 30, 40, 50};
    double avg; int mx, mn;
    computeStats(d, avg, mx, mn);
    cout << "avg=" << avg << " max=" << mx << " min=" << mn << endl;
    return 0;
}
